template.go

```diff
@@ -398,11 +399,24 @@ func generateFile(config Config, containers Context) bool {
 	}
 
 	var buf bytes.Buffer
-	multiwriter := io.MultiWriter(dest, &buf)
-	err = tmpl.ExecuteTemplate(multiwriter, filepath.Base(templatePath), &filteredContainers)
+	bw := bufio.NewWriter(&buf)
```

Comment:
This seems to have broken the code that compares if the generated file is different from the existing one.  

Running `docker-gen -skip-blank-lines ./templates/nginx.tmpl /tmp/nginx` always generates a new `/tmp/nginx` even if the contents have not changed.    The issue is that `bytes.Buffer` clears it's buffer after `WriteTo` is called on L418.  So the comparison on L437 always fails.

See: http://play.golang.org/p/tcz4g_VHZ_